packages/block-editor/src/layouts/flex.js

```diff
@@ -66,16 +66,19 @@ export default {
 		onChange,
 		layoutBlockSupport = {},
 	} ) {
-		const { allowOrientation = true } = layoutBlockSupport;
+		const { allowOrientation = true, allowJustification = true } =
+			layoutBlockSupport;
 		return (
 			<>
 				<Flex>
-					<FlexItem>
-						<FlexLayoutJustifyContentControl
-							layout={ layout }
-							onChange={ onChange }
-						/>
-					</FlexItem>
+					{ allowJustification && (
+						<FlexItem>
+							<FlexLayoutJustifyContentControl
+								layout={ layout }
+								onChange={ onChange }
+							/>
+						</FlexItem>
+					) }
 					<FlexItem>
 						{ allowOrientation && (
```

Comment:
Can we not render `FlexItem` if `! allowOrientation`? It was from before, but we can update it here.

Reply:
Yes, updated!